axelrod/tests/strategies/test_gradualkiller.py

```diff
@@ -19,78 +19,59 @@ class TestGradualKiller(TestPlayer):
+                         expected_actions=expected)
 
     def test_effect_of_strategy_with_history_CD(self):
         """Continues with C if opponent played CD on 6 and 7."""
-        P1 = axelrod.GradualKiller()
-        P2 = axelrod.Player()
-        P1.history = [D, D, D, D, D, C, C]
-        P2.history = [C, C, C, C, C, C, D]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C]
-        P2.history = [C, C, C, C, C, C, D, D]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C, C]
-        P2.history = [C, C, C, C, C, C, D, D, C]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C, C, C]
-        P2.history = [C, C, C, C, C, C, D, D, C, C]
-        self.assertEqual(P1.strategy(P2), 'C')
+        expected, opponent = self.get_actions_for_test(D, [C, D], C)
+        self.versus_test(axl.MockPlayer(actions=opponent),
+                         expected_actions=expected)
 
     def test_effect_of_strategy_with_history_DC(self):
         """Continues with C if opponent played DC on 6 and 7."""
-        P1 = axelrod.GradualKiller()
-        P2 = axelrod.Player()
-        P1.history = [D, D, D, D, D, C, C]
-        P2.history = [C, C, C, C, C, D, C]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C]
-        P2.history = [C, C, C, C, C, D, C, C]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C, C]
-        P2.history = [C, C, C, C, C, D, C, C, D]
-        self.assertEqual(P1.strategy(P2), 'C')
-        P1.history = [D, D, D, D, D, C, C, C, C, C]
-        P2.history = [C, C, C, C, C, D, C, C, D, C]
-        self.assertEqual(P1.strategy(P2), 'C')
+        expected, opponent = self.get_actions_for_test(D, [D, C], C)
+        self.versus_test(axl.MockPlayer(actions=opponent),
+                         expected_actions=expected)
 
     def test_effect_of_strategy_with_history_DD(self):
         """Continues with D if opponent played DD on 6 and 7."""
-        P1 = axelrod.GradualKiller()
-        P2 = axelrod.Player()
-        P1.history = [D, D, D, D, D, C, C]
-        P2.history = [C, C, C, C, C, D, D]
-        self.assertEqual(P1.strategy(P2), 'D')
-        P1.history = [D, D, D, D, D, C, C, D]
-        P2.history = [C, C, C, C, C, D, D, C]
-        self.assertEqual(P1.strategy(P2), 'D')
-        P1.history = [D, D, D, D, D, C, C, D, D]
-        P2.history = [C, C, C, C, C, D, D, C, C]
-        self.assertEqual(P1.strategy(P2), 'D')
-        P1.history = [D, D, D, D, D, C, C, D, D, D]
-        P2.history = [C, C, C, C, C, D, D, C, C, D]
-        self.assertEqual(P1.strategy(P2), 'D')
+        expected, opponent = self.get_actions_for_test(C, [D, D], D)
+        self.versus_test(axl.MockPlayer(actions=opponent),
+                         expected_actions=expected)
+
+    def get_actions_for_test(self, opponent_start, action_six_and_seven,
```

Comment:
Unless they're truly necessary it's nicer to not make use of these helper functions and write out tests explicitly: otherwise technically we also want a test for this helper function. 

If we do actually need this helper function then it should certainly be better documented.